packages/core/src/mastra/index.ts

```diff
@@ -684,6 +699,28 @@ do:
     return workflow;
   }
 
+  __hasInternalWorkflow(id: string): boolean {
+    return Object.values(this.#internalMastraWorkflows).some(workflow => workflow.id === id);
+  }
+
+  __getInternalWorkflow(id: string): Workflow {
+    const workflow = Object.values(this.#internalMastraWorkflows).find(a => a.id === id);
+    if (!workflow) {
+      throw new MastraError({
+        id: 'MASTRA_GET_INTERNAL_WORKFLOW_BY_ID_NOT_FOUND',
+        domain: ErrorDomain.MASTRA,
+        category: ErrorCategory.SYSTEM,
+        text: `Workflow with id ${String(id)} not found`,
+        details: {
+          status: 404,
+          workflowId: String(id),
+        },
+      });
+    }
+
+    return workflow;
+  }
+
```

Comment:
would be cool to add a `__getInternalWorkflow()` or `__registerInternalWorkflow()` method that could be used in scorers and other systems to register workflows as internal.